scaladoc/src/dotty/tools/scaladoc/snippets/SnippetCompiler.scala

```diff
@@ -0,0 +1,77 @@
+package dotty.tools.scaladoc
+package snippets
+
+import dotty.tools.io.{AbstractFile, VirtualDirectory}
+import dotty.tools.dotc.interactive.InteractiveDriver
+import dotty.tools.dotc.interactive.Interactive
+import dotty.tools.dotc.interactive.InteractiveCompiler
+import dotty.tools.dotc.core.Contexts.Context
+import dotty.tools.dotc.config.Settings.Setting._
+import dotty.tools.dotc.interfaces.SourcePosition
+import dotty.tools.dotc.ast.Trees.Tree
+import dotty.tools.dotc.interfaces.{SourceFile => ISourceFile}
+import dotty.tools.dotc.reporting.{ Diagnostic, StoreReporter }
+import dotty.tools.dotc.parsing.Parsers.Parser
+import dotty.tools.dotc.{ Compiler, Run }
+import dotty.tools.io.{AbstractFile, VirtualDirectory}
+import dotty.tools.repl.AbstractFileClassLoader
+import dotty.tools.dotc.util.SourceFile
+
+class SnippetCompiler(
+  classpath: String = System.getProperty("java.class.path"), //Probably needs to be done better
+  val scalacOptions: String = "",
+  target: AbstractFile = new VirtualDirectory("(memory)")
+):
+
+  private def newDriver: InteractiveDriver = {
```

Comment:
In future we can probably stop after the given phases since we probably do not need to produce bytecode.